### ISSUE TITLE
test-process-active-wraps.js fails on windows

### ISSUE THREAD

**Issue by misterdjules** [joyent/node#8986]
It seems that commit b636ba8186d191c52ee36f2f2b1aebbbb4d95575 introduced the following regression on Windows:
```
>.\Release\node.exe test\simple\test-process-active-
wraps.js

assert.js:98
  throw new assert.AssertionError({
        ^
AssertionError: 1 == 0
    at Immediate._onImmediate (c:\Users\jgilli\dev\node-bis\test\simple\test-pro
cess-active-wraps.js:69:16)
    at processImmediate [as _immediateCallback] (timers.js:340:17)

>
```

@cjihrig Would you mind taking a look?

**Comment by cjihrig** [comment]
The failure is apparently related to the removal of [these lines](https://github.com/joyent/node/blob/8cfbeed27ab98dffff68c13188d4f3dc1a506c3c/lib/net.js#L901-L904). I'm not sure why the extra handle hangs around on Windows when a DNS lookup occurs, but I will restore the relevant code as a bandaid.

**Comment by cjihrig** [comment]
Closed via 1fad3730c2d6130c13b46ad78ded808fe9bb4c10
